source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java

```diff
@@ -114,8 +128,285 @@ static void setup() {
+     */
+//    @Test
+//    void TestFrogHasAnimation() {
+//        Assertions.assertTrue(friendlyFrog.getComponent(AnimationRenderComponent.class).hasAnimation("wait"),
+//                "Frog should have idle animation.");
+//    }
+
+    /**
+     * Tests that the friendly frog is a friendly NPC meaning it won't attack players.
+     */
+    @Test
+    void TestFrogIsFriendly() {
+        Assertions.assertNotNull(friendlyFrog.getComponent(FriendlyNPCAnimationController.class),
+                "Frog should have a friendly AI controller.");
+    }
+
+    /**
+     * Tests that the friendly frog is in the correct spot when placed.
+     */
+    @Test
+    void TestFrogSetPosition() {
+        Vector2 pos = new Vector2(0f, 0f);
+        friendlyFrog.setPosition(pos);
+
+        Assertions.assertEquals(pos, friendlyFrog.getPosition());
+    }
+
+
+    /**
+     * Tests the initialization of a friendly monkey by checking its creation, name, type,
+     * and the presence of necessary components.
+     */
+    @Test
+    void testMonkeyInitialisation() {
+        // Test creation of the friendlyMonkey
+        Assertions.assertNotNull(friendlyMonkey, "Monkey should not be null.");
+
+        // Test that the friendlyMonkey has the correct name
+        String name = configs.friendlyMonkey.getAnimalName();
+        Assertions.assertEquals("Monkey", name);
+
+        // Test that the friendlyMonkey is an Entity
+        Assertions.assertEquals(Entity.class, friendlyMonkey.getClass());
+
+        // Test that the friendlyMonkey has a PhysicsComponent
+        Assertions.assertNotNull(friendlyMonkey.getComponent(PhysicsComponent.class));
+
+        // Test that the friendlyMonkey has a PhysicsMovementComponent
+        Assertions.assertNotNull(friendlyMonkey.getComponent(PhysicsMovementComponent.class));
+
+        // Test that the friendlyMonkey has a ColliderComponent
+        Assertions.assertNotNull(friendlyMonkey.getComponent(ColliderComponent.class));
+
+        // Test that the friendlyMonkey has a ConfigComponent
+        Assertions.assertNotNull(friendlyMonkey.getComponent(ConfigComponent.class));
+    }
+
+    /**
+     * Tests that the friendly monkey has the correct sound path.
+     * //TODO:Change when sound path is updated
+     */
+    @Test
+    void TestMonkeyHasCorrectSoundPath() {
+        String[] sound = configs.friendlyMonkey.getSoundPath();
+        Assertions.assertNotNull(sound);
+        Assertions.assertArrayEquals(new String[]{"sounds/monkey.wav"}, sound);
+    }
+
+    /**
+     * Tests that the friendly monkey has the correct base hint.
+     */
+    @Test
+    void TestMonkeyHasCorrectBaseHint() {
+        String[][] baseHint = configs.friendlyMonkey.getBaseHint();
+        assertNotNull(baseHint);
+        Assertions.assertArrayEquals(new String[][]{{"Welcome to Animal Kingdom!","I am Max the Monkey."}},
+                baseHint);
+    }
+
+    /**
+     * Tests that the friendly monkey has a wait animation.
+     */
```

Comment:
Why is this test commented out? Please either make it work with the current game or remove it.